### PR TITLE
Schematic Share system

### PR DESCRIPTION
**Pull request by me4502**
This PR adds a command in game, that shares a schematic with a URL.

`/schem share`.

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java [hunk 1]
```diff
@@ -372,6 +408,49 @@ public Void call() throws Exception {
...
+                baos = closer.register(new ByteArrayOutputStream());
+                BufferedOutputStream bos = closer.register(new BufferedOutputStream(baos));
+                ClipboardWriter writer = closer.register(format.getWriter(bos));
+                writer.write(target);
+            } catch (IOException e) {
+                throw new CommandException(TextComponent.of(e.getMessage()), e, ImmutableList.of());
+            }
+
+            EngineHubPaste pasteService = new EngineHubPaste();
+            return pasteService.paste(new String(Base64.getEncoder().encode(baos.toByteArray()), Charsets.UTF_8)).call();
```
Comment (kashike) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java:450]:
```suggestion
            return pasteService.paste(new String(Base64.getEncoder().encode(baos.toByteArray()), StandardCharsets.UTF_8)).call();
```
No need to use Guava, the JDK has this.

**Review by kashike — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java [hunk 2]
```diff
@@ -146,7 +152,7 @@ public void load(Actor actor, LocalSession session,
 
     @Command(
         name = "save",
-        desc = "Save a schematic into your clipboard"
+        desc = "Save a schematic of your clipboard"
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java:155]:
```suggestion
        desc = "Save your clipboard into a schematic file"
```

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java [hunk 3]
```diff
@@ -205,6 +211,39 @@ public void save(Actor actor, LocalSession session,
                 .buildAndExec(worldEdit.getExecutorService());
     }
 
+    @Command(
+        name = "share",
+        desc = "Share a schematic into your clipboard"
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java:216]: smh you corrected the `save` and then made the same mistake here
```suggestion
        desc = "Share your clipboard as a schematic online"
```

File: worldedit-core/src/main/java/com/sk89q/worldedit/util/paste/PasteMetadata.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util.paste;
+
+public class PasteMetadata {
+    public PasteMetadata() {
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/util/paste/PasteMetadata.java:23]: This is kinda pointless. Use the bits of syntax sugar Java gives you :P

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java [hunk 5]
```diff
@@ -372,6 +411,55 @@ public Void call() throws Exception {
...
+        SchematicShareTask(Actor actor, ClipboardFormat format, ClipboardHolder holder, String name) {
+            this.actor = actor;
+            this.format = format;
+            this.holder = holder;
+            this.name = name;
+        }
+
+        @Override
+        public URL call() throws Exception {
+            Clipboard clipboard = holder.getClipboard();
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java:429]: Can we merge the saving code here and with `save` into a single method? DRY applies here, especially with how much custom logic is part of saving.

**Review by octylFractal — CHANGES_REQUESTED** [review]

**Review by octylFractal — CHANGES_REQUESTED** [review]

**Review by octylFractal — APPROVED** [review]

**Pull request merged**